kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/Aes.java

```diff
@@ -20,22 +20,29 @@
 
 public class Aes implements CipherManager {
 
-    public static final Aes AES_256_GCM_128 = new Aes("AES_256/GCM/NoPadding", (byte) 0, CipherSpec.AES_256_GCM_128);
+    public static final Aes AES_256_GCM_128 = new Aes("AES/GCM/NoPadding", 256, (byte) 0, CipherSpec.AES_256_GCM_128);
 
     private static final int IV_SIZE_BYTES = 12;
     private static final int TAG_LENGTH_BITS = 128;
     private final String transformation;
     private final byte serializedId;
     private final CipherSpec spec;
     private final SecureRandom rng;
+    private final int numKeyBits;
 
-    private Aes(String transformation, byte serializedId, CipherSpec spec) {
+    private Aes(String transformation, int numKeyBits, byte serializedId, CipherSpec spec) {
         this.transformation = transformation;
+        this.numKeyBits = numKeyBits;
```

Comment:
Should we be validating that numKeyBits is a positive non zero number?

Reply:
It's a private constructor, so there's very marginal value. But sure why not.

```suggestion
        if (numKeyBits < 0) {
            throw new IllegalArgumentException();
        }
        this.numKeyBits = numKeyBits;
```